homeassistant/components/systemmonitor/sensor.py

```diff
@@ -334,39 +340,84 @@ async def async_setup_platform(
+                )
+                entities.append(
+                    SystemMonitorSensor(
+                        sensor_registry, sensor_description, entry.entry_id, argument
+                    )
+                )
+            continue
 
-    for resource in config[CONF_RESOURCES]:
-        type_ = resource[CONF_TYPE]
-        # Initialize the sensor argument if none was provided.
-        # For disk monitoring default to "/" (root) to prevent runtime errors, if argument was not specified.
-        if CONF_ARG not in resource:
-            argument = ""
-            if resource[CONF_TYPE].startswith("disk_"):
-                argument = "/"
-        else:
-            argument = resource[CONF_ARG]
+        if _type in NETWORK_TYPES:
+            arguments = await hass.async_add_executor_job(get_all_network_interfaces)
+            for argument in arguments:
+                sensor_registry[(_type, argument)] = SensorData(
+                    argument, None, None, None, None
+                )
+                entities.append(
+                    SystemMonitorSensor(
+                        sensor_registry, sensor_description, entry.entry_id, argument
+                    )
+                )
+            continue
 
         # Verify if we can retrieve CPU / processor temperatures.
         # If not, do not create the entity and add a warning to the log
         if (
-            type_ == "processor_temperature"
+            _type == "processor_temperature"
             and await hass.async_add_executor_job(_read_cpu_temperature) is None
         ):
             _LOGGER.warning("Cannot read CPU / processor temperature information")
             continue
 
-        sensor_registry[(type_, argument)] = SensorData(
-            argument, None, None, None, None
-        )
+        if _type == "process":
+            entries: list[dict[str, str]] = entry.options.get(SENSOR_DOMAIN, [])
+            for _entry in entries:
+                for _, argument in _entry.items():
+                    sensor_registry[(_type, argument)] = SensorData(
+                        argument, None, None, None, None
+                    )
+                    entities.append(
+                        SystemMonitorSensor(
+                            sensor_registry,
+                            sensor_description,
+                            entry.entry_id,
+                            argument,
+                        )
+                    )
+            continue
+
+        sensor_registry[(_type, "")] = SensorData("", None, None, None, None)
         entities.append(
-            SystemMonitorSensor(sensor_registry, SENSOR_TYPES[type_], argument)
+            SystemMonitorSensor(sensor_registry, sensor_description, entry.entry_id, "")
         )
 
-    scan_interval = config.get(CONF_SCAN_INTERVAL, DEFAULT_SCAN_INTERVAL)
+    scan_interval = DEFAULT_SCAN_INTERVAL
     await async_setup_sensor_registry_updates(hass, sensor_registry, scan_interval)
```

Comment:
Side note: The sensor update seems really inefficient now as it updates all sensor registry items even though the corresponding entities may be disabled. Maybe we already have plans to refactor that later?